src/reducers/auth.js

```diff
@@ -0,0 +1,56 @@
+import axios from 'axios'
+
+/*
+  ACTION TYPE
+ */
+const SET = 'SET_CURRENT_USER'
+const REMOVE = 'REMOVE_CURRENT_USER'
+
+/*
+  ACTION CREATOR
+ */
+const setCurrUser = user => ({ type: SET, user })
+const removeCurrUser = () => ({ type: REMOVE })
+
+/*
+  REDUCER
+ */
+export default (currentUser = {}, action) => {
+  switch (action.type) {
+    case SET:
+      return action.user
+    case REMOVE:
+      return {}
+    default:
+      return currentUser;
+  }
+}
+
+/*
+  THUNK
+ */
+const login = (credentials, history) => dispatch => {
+  return axios.post('api/auth', credentials)
+    .then(result => result.data)
+    .then(user => {
```

Comment:
If I'm being picky, you don't really need this result => result.data line, you could just do dispatch(setCurrUser(result.data))